### PR TITLE
Added response type docs for prompts

### PR DESCRIPTION
**Pull request by AlecAivazis**
@ahmetb, I added response type documentation for the various prompts in order to address your concerns in #82 and #81. Please let me know if there is something better you would like to see.

### CODE REVIEW

**Comment by ahmetb** [comment]
This is great. Thanks. 

Consider using "go test" ExampleXxx format to provide examples easily viewable in the generated godoc.

File: confirm.go [hunk 1]
```diff
@@ -104,8 +104,14 @@ func (c *Confirm) getBool(showHelp bool) (bool, error) {
...
 
-// Prompt prompts the user with a simple text field and expects a reply followed
-// by a carriage return.
+/*
+Prompt prompts the user with a simple text field and expects a reply followed
+by a carriage return.
+
+	likesPie := false
+	prompt := &survey.Confirm{ Message: "What is your name?" }
+	survey.AskOne(prompt, &name, nil)
```
Comment (coryb) [inline confirm.go]: This one looks like a bad copy/paste, correct the prompt to something like `Do you like Pie?`, and use the `likePies` variable.

**Comment by coryb** [comment]
I was going to suggest getting ExampleXxx tests working and started playing with it myself, but it is tricky if we wanted to use the `// Output: ` to test the output because of the terminal control characters that are printed.  We could make some Examples that just test syntax like this:
```
func ExampleConfirm_Prompt() {
    likesPie := false
    prompt := &Confirm{Message: "Do you like pie?"}
    AskOne(prompt, &likesPie, nil)
}
```
With this the godocs will have a magic example that shows up.

File: confirm.go (hunk 1 above)
Comment (AlecAivazis) [inline confirm.go]: Nice catch - fixing it now on a separate branch.

**Comment by AlecAivazis** [comment]
@coryb Sorry for not waiting for the review, I was feeling bad that there were no useful docs on the godoc page and figured we could get your changes in after the fact.

I fixed the typo on another branch, feel free to push onto that if you end up getting the examples going, otherwise I can try to get something together tonight when I get back from dinner.

**Comment by coryb** [comment]
n/p, I was busy most of the weekend.  I probably wont have time to do get to the examples really in the next week, so if you feel like making the changes go ahead :)

**Comment by AlecAivazis** [comment]
So I spent some time thinking about this further and I'm not sure what I think about the documentation being under `Prompt`, since that's sort of an implementation detail. I couldn't figure out how to provide a type-wide example, so I think I am in favor of keeping it how it is now - thoughts?

**Pull request merged**